src/main/java/jenkins/plugins/http_request/HttpRequest.java

```diff
@@ -176,16 +202,23 @@ public String getValidResponseContent() {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-        defineDefaultConfigurations();
-
+    public void perform(Run<?,?> run, FilePath workspace, Launcher launcher, TaskListener listener)
```

Comment:
If you want to keep this public method compatibility you have to leave it and call the new one with something like:

public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
    return perform(build, build.getWorkspace(), launcher, listener);
}